pkg/cassandra/config_premarshal.go

```diff
@@ -66,19 +72,87 @@ func preMarshalConfig(val reflect.Value, version *semver.Version) (map[string]in
 	return out, nil
 }
 
-func getFieldValue(fieldVal reflect.Value, structType reflect.Type, field reflect.StructField) (interface{}, error) {
+// getFieldValueRecursive returns the value of a complex field annotated with the "recurse" option,
```

Comment:
Do we need the `recurse` option? Don't we always want to recuse into complex fields?

Reply:
No, not always. `resource.Quantity` for instance should not be recursed but serialized as a scalar.